pkg/image/registry/imagestreamtag/rest.go

```diff
@@ -315,34 +315,47 @@ func (r *REST) Delete(ctx apirequest.Context, id string, options *metav1.DeleteO
 		return nil, false, err
 	}
 
-	stream, err := r.imageStreamRegistry.GetImageStream(ctx, name, &metav1.GetOptions{})
-	if err != nil {
-		return nil, false, err
-	}
+	for i := 10; i > 0; i-- {
+		stream, err := r.imageStreamRegistry.GetImageStream(ctx, name, &metav1.GetOptions{})
+		if err != nil {
+			return nil, false, err
+		}
+		if pre := options.Preconditions; pre != nil {
+			if pre.UID != stream.UID {
+				return nil, false, kapierrors.NewConflict(imageapi.Resource("imagestreamtags"), id, fmt.Errorf("the UID precondition was not met"))
+			}
+		}
 
-	notFound := true
+		notFound := true
 
-	// Try to delete the status tag
-	if _, ok := stream.Status.Tags[tag]; ok {
-		delete(stream.Status.Tags, tag)
-		notFound = false
-	}
+		// Try to delete the status tag
+		if _, ok := stream.Status.Tags[tag]; ok {
+			delete(stream.Status.Tags, tag)
```

Comment:
It looks like ` // The tag history remains intact and is not deleted.` in the godoc does not apply anymore.